examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java

```diff
@@ -167,35 +170,84 @@ public static void postProcessAndPrintOutput(CommandLine cmd, String output, Str
                     if (node.isEmpty()) {
                         System.out.println(output);
                         return;
                     } else {
-                        List<String> columns = getColumns(node.get(0),at);
-                        int totalWidth = 190;
-                        int nameColumnWidth = 12;
-                        int columnWidth = (totalWidth-nameColumnWidth)/(columns.size() - 1);
+                        List<String> fields = getFields(node.get(0));
+
+                        int[] columnWidths = new int[fields.size()];
+                        boolean[] fixedWidthColumns = new boolean[fields.size()];
+                        int maxLineWidth = 0;
+
+                        for (int columnIndex = 0; columnIndex < fields.size(); columnIndex++) {
+                            columnWidths[columnIndex] = fields.get(columnIndex).length();
+                        }
+
+                        for (JsonNode jsonNode : node) {
+                            int lineWidth = 1;
+                            for (int columnIndex = 0; columnIndex < fields.size(); columnIndex++) {
+                                String fieldName = fields.get(columnIndex);
+                                JsonNode value = jsonNode.get(fieldName);
+                                String valueString = value.isTextual() ? value.asText() : value.toString();
+
+                                columnWidths[columnIndex] = Math.max(columnWidths[columnIndex], valueString.length());
+                                lineWidth += valueString.length() + 1;
+
+                                String fieldNameLowerCase = fieldName.toLowerCase();
+                                if (fieldNameLowerCase.equals("name") || fieldNameLowerCase.endsWith("id")) {
+                                    fixedWidthColumns[columnIndex] = true;
+                                }
+
+                            }
+                            maxLineWidth = Math.max(maxLineWidth, lineWidth);
+                        }
+
+                        if (maxLineWidth >= outputWidth) {
+                            int widthRemaining = outputWidth - fields.size();
+
+                            int fixedWidthColumnCount = 0;
+                            for (int i = 0; i < fields.size(); i++) {
+                                widthRemaining -= fixedWidthColumns[i] ? columnWidths[i] : 0;
+                                if (fixedWidthColumns[i]) {
+                                    fixedWidthColumnCount++;
+                                }
+                            }
+                            int columnWidth = (widthRemaining) / (fields.size() - fixedWidthColumnCount);
+
+                            for (int i = 0; i < fields.size(); i++) {
+                                if (!fixedWidthColumns[i]) {
+                                    columnWidths[i] = columnWidth;
+                                }
+                            }
+                        }
```

Comment:
This looks great, could you please extract out these blocks of logic into separate functions to make this more readable, I think `postProcessAndPrintOutput` has become too big